README.md

```diff
@@ -73,26 +104,150 @@ The system uses dedicated queues for managing different job phases:
+- Prover Service (ATLANTIC)
+- MongoDB for job management
+- AWS services (or Localstack for local development):
+  - SQS for queues
+  - S3 for data storage
+  - SNS for alerts
+  - EventBridge for scheduling
+
+> 🚨 **Important Note**: Currently, Madara doesn't support the `get_storage_proof` endpoint.
+>
+> Therefore, you need to run Pathfinder alongside Madara:
+>
+> - Madara will run in sequencer mode
+> - Pathfinder will sync with Madara
+> - The orchestrator will use Pathfinder's RPC URL for SNOS and state update fetching
+>
+> This setup ensures proper functionality until the `get_storage_proof` endpoint is implemented in Madara.
 
-1. **Madara Node**
+## 🚀 Installation & Setup
 
-   - Required for block processing
-   - Follow setup instructions at [Madara Documentation](https://github.com/madara-alliance/madara)
+### Building from Source
 
-2. **Prover Service**
+1. **Install System Dependencies**
 
-   - ATLANTIC must be running
+   ```bash
+   # Ubuntu/Debian
+   sudo apt-get update
+   sudo apt install build-essential openssl pkg-config libssl-dev
+   sudo apt install python3.9 python3.9-venv python3.9-distutils libgmp-dev python3.9-dev
 
-3. **Infrastructure Dependencies**
-   - MongoDB for job management
-   - AWS services (or Localstack for local development):
-     - SQS for queues
-     - S3 for data storage
-     - SNS for alerts
-     - EventBridge for scheduling
+   # For macOS
+   brew install openssl pkg-config gmp python@3.9
+   ```
 
-## 🚀 Installation & Setup
+2. **Install Rust** (Cross-platform)
+
+   ```bash
+   curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+   source ~/.bashrc  # Or source ~/.zshrc for macOS
+   ```
+
+3. **Clone Repository**
+
+   ```bash
+   git clone https://github.com/madara-alliance/madara-orchestrator.git
+   cd madara-orchestrator
+   git submodule update --init
+   ```
+
+4. **Build SNOS**
+
+   ```bash
+   make snos
+   ```
+
+5. **Build Project**
+
+   ```bash
+   cargo build --release
+   ```
+
+### Local Development Setup
+
+1. **Install Docker** (Cross-platform)
+   Follow the official installation guides:
+
+   - [Ubuntu Installation Guide](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository)
+   - [macOS Installation Guide](https://docs.docker.com/desktop/install/mac-install/)
+
+2. **Install Foundry** (Cross-platform)
+
+   ```bash
+   curl -L https://foundry.paradigm.xyz | bash
+   foundryup
+   ```
+
+3. **Start Local Services**
+
+   ```bash
+   # Start MongoDB
+   docker run -d -p 27017:27017 mongo
+
+   # Start Localstack
+   docker run -d -p 4566:4566 localstack/localstack@sha256:763947722c6c8d33d5fbf7e8d52b4bddec5be35274a0998fdc6176d733375314
+
+   # Start Anvil in a separate terminal
+   anvil --block-time 1
+   ```
+
+4. **Setup Mock Proving Service**
+
+   ```bash
+   # Create a simple HTTP server that always returns 200
+   # Example using Python
+   python3 -m http.server 8888 &
```

Comment:
oh interesting, so we don't need that mock prover service arun wrote?

Reply:
it was not working, changed this